outboundSender.go

```diff
@@ -448,13 +451,29 @@ func (obs *CaduceusOutboundSender) isValidTimeWindow(now, dropUntil, deliverUnti
 	return true
 }
 
+func (obs *CaduceusOutboundSender) Empty() {
+	for obs.queueEmpty == false {
+		select {
+		case <-obs.queue:
+			obs.queueDepthGauge.Add(-1.0)
+			continue
+		default:
+			obs.queueEmpty = true
+		}
+	}
+	return
+}
+
 func (obs *CaduceusOutboundSender) dispatcher() {
 	defer obs.wg.Done()
 
 	for msg := range obs.queue {
 		obs.queueDepthGauge.Add(-1.0)
 
 		obs.mutex.RLock()
+		if obs.queueEmpty == false {
```

Comment:
```suggestion
		if !obs.queueEmpty {
```
this is a superficial thing, feel free to ignore.